test/test_build.py

```diff
@@ -196,5 +198,24 @@ def test_build_expo_backoff_when_luigi_failed_due_to_locked_task(self):
         gokart.build(_FailThreeTimesAndSuccessTask(), reset_register=False)
 
 
+class TestBuildFailedAndSchedulingFailed(unittest.TestCase):
+    def test_build_raises_exception_on_failed_and_scheduling_failed(self):
+        """Test that build() raises GokartBuildError when FAILED_AND_SCHEDULING_FAILED occurs"""
+
+        # Create a mock result object with FAILED_AND_SCHEDULING_FAILED status
+        class MockResult:
+            def __init__(self):
+                self.status = luigi.LuigiStatusCode.FAILED_AND_SCHEDULING_FAILED
```

Comment:
It's good to test the error cases (FAILED_AND_SCHEDULING_FAILED)! It would be even better to add tests for the statuses that should succeed in the normal flow.